test/nest_test.cpp

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License Version 2.0 with LLVM Exceptions
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *   https://llvm.org/LICENSE.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <unifex/nest.hpp>
+
+#include <unifex/just.hpp>
+#include <unifex/sync_wait.hpp>
+#include <unifex/v1/async_scope.hpp>
+#include <unifex/v2/async_scope.hpp>
+
+#include <gtest/gtest.h>
+
+namespace {
+
+struct tag_invocable_scope {
+  template <typename Sender>
+  friend auto tag_invoke(
+      unifex::tag_t<unifex::nest>,
+      Sender&& sender,
+      tag_invocable_scope& scope) noexcept {
+    scope.invoked = true;
+    return static_cast<Sender&&>(sender);
+  }
+
+  bool invoked{false};
+};
+
+struct member_invocable_scope {
```

Comment:
What happens if you provide both `tag_invoke(unifex::tag_t<unifex::nest>)` and `nest()`?